chainer/optimizer.py

```diff
@@ -466,176 +457,9 @@ def use_cleargrads(self, use=True):
-
-    def __init__(self, threshold):
-        self.threshold = threshold
-
-    def __call__(self, opt):
-        norm = numpy.sqrt(_sum_sqnorm([p.grad for p in opt.target.params()]))
-        rate = self.threshold / norm
-        if rate < 1:
-            for param in opt.target.params():
-                grad = param.grad
-                with cuda.get_device(grad):
-                    grad *= rate
-
-
-class GradientNoise(object):
-    """Optimizer hook function for adding gradient noise.
-
-    This hook function simply adds noise generated by the ``noise_func``
-    to the gradient. By default it adds time-dependent annealed Gaussian
-    noise to the gradient at every training step:
-
-    .. math::
-
-        g_t \\leftarrow g_t + N(0, \\sigma_t^2)
-
-    where
-
-    .. math::
-
-        \\sigma_t^2 = \\frac{\\eta}{(1+t)^\\gamma}
-
-    with :math:`\\eta` selected from {0.01, 0.3, 1.0} and
-    :math:`\\gamma = 0.55`.
-
-    Args:
-        eta (float): Parameter that defines the scale of the noise, which for
-            the default noise function is recommended to be either 0.01, 0.3
-            or 1.0.
-        noise_func (function): Noise generating function which by default
-            is given by `Adding Gradient Noise Improves Learning for Very Deep\
-            Networks <https://arxiv.org/pdf/1511.06807>`_.
-    """
-    name = 'GradientNoise'
-
-    def __init__(self, eta, noise_func=exponential_decay_noise):
-        self.eta = eta
-        self.noise_func = noise_func
-
-    def kernel(self):
-        return cuda.elementwise(
-            'T noise', 'T g', 'g += noise', 'gradient_noise')
-
-    def __call__(self, opt):
-        for param in opt.target.params():
-            g = param.grad
-            xp = cuda.get_array_module(g)
-            with cuda.get_device(g) as dev:
-                noise = self.noise_func(xp, g.shape, g.dtype, self, opt)
-                if int(dev) == -1:
-                    g += noise
-                else:
-                    self.kernel()(noise, g)
-
-
-class GradientHardClipping(object):
-    """Optimizer hook function for gradient clipping.
-
-    This hook function clips all gradient arrays to be within a lower and upper
-    bound.
-
-    Args:
-        lower_bound (float): The lower bound of the gradient value.
-        upper_bound (float): The upper bound of the gradient value.
-
-    Attributes:
-        lower_bound (float): The lower bound of the gradient value.
-        upper_bound (float): The upper bound of the gradient value.
-
-    """
-    name = 'GradientHardClipping'
-
-    def __init__(self, lower_bound, upper_bound):
-        self.lower_bound = lower_bound
-        self.upper_bound = upper_bound
-
-    def __call__(self, opt):
-        xp = opt.target.xp
-        for param in opt.target.params():
-            grad = param.grad
-            with cuda.get_device(grad):
-                xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)
+#  Backward compatibility
```

Comment:
Should we use some hack to hook deprecation warning like [this](http://stackoverflow.com/a/922693)?